vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/per_day_frequency_cap_unittest.cc

```diff
@@ -0,0 +1,178 @@
+
+const uint64_t kSecondsPerDay = base::Time::kSecondsPerHour *
+    base::Time::kHoursPerDay;
+
+class BraveAdsPerDayFrequencyCapTest : public ::testing::Test {
+ protected:
+  BraveAdsPerDayFrequencyCapTest()
+  : mock_ads_client_(std::make_unique<MockAdsClient>()),
+    ads_(std::make_unique<AdsImpl>(mock_ads_client_.get())) {
+    // You can do set-up work for each test here
+  }
+
+  ~BraveAdsPerDayFrequencyCapTest() override {
+    // You can do clean-up work that doesn't throw exceptions here
+  }
+
+  // If the constructor and destructor are not enough for setting up and
+  // cleaning up each test, you can use the following methods
+
+  void SetUp() override {
+    // Code here will be called immediately after the constructor (right before
+    // each test)
+
+    auto callback = std::bind(
+        &BraveAdsPerDayFrequencyCapTest::OnAdsImplInitialize, this, _1);
+    ads_->Initialize(callback);
+
+    client_mock_ = std::make_unique<ClientMock>(ads_.get(),
+        mock_ads_client_.get());
+    frequency_capping_ = std::make_unique<FrequencyCapping>(client_mock_.get());
+    exclusion_rule_ = std::make_unique<PerDayFrequencyCap>(
+        frequency_capping_.get());
+  }
+
+  void OnAdsImplInitialize(const Result result) {
+    EXPECT_EQ(Result::SUCCESS, result);
+  }
+
+  void TearDown() override {
+    // Code here will be called immediately after each test (right before the
+    // destructor)
+  }
+
+  std::unique_ptr<MockAdsClient> mock_ads_client_;
+  std::unique_ptr<AdsImpl> ads_;
+
+  std::unique_ptr<ClientMock> client_mock_;
+  std::unique_ptr<FrequencyCapping> frequency_capping_;
+  std::unique_ptr<PerDayFrequencyCap> exclusion_rule_;
+  AdInfo ad_info_;
+};
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdAllowedWhenNoAds) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 2;
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdAllowedBelowDailyCap) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 2;
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, 0, 1);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdAllowedWithAdOutsideDayWindow) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 2;
+  // Now
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, 0, 1);
+  // 24hrs + 1s ago
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, -kSecondsPerDay, 1);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdExcludedAboveDailyCap1) {
```

Comment:
nit: AdExcludedAboveDailyCapOf1

Reply:
This one isn't 'a cap of 1' but a variant on the test name itself... `AdExcludedAboveDailyCap1` & `AdExcludedAboveDailyCap2` happy to change it to something better but no inspiration as both tests are very similar. Ideas welcome!

Reply:
Please use descriptive names, even if long explaining the edge cases

Reply:
Will rethink 👍

Reply:
Done